aws/data_source_aws_launch_configuration.go

```diff
@@ -0,0 +1,221 @@
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+
+			"associate_public_ip_address": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+
+			"spot_price": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"ebs_optimized": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+
+			"placement_tenancy": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"enable_monitoring": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+
+			"ebs_block_device": {
+				Type:     schema.TypeSet,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"delete_on_termination": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+
+						"device_name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+
+						"iops": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+
+						"snapshot_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+
+						"volume_size": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+
+						"volume_type": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+
+						"encrypted": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+					},
+				},
+			},
+
+			"ephemeral_block_device": {
+				Type:     schema.TypeSet,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"device_name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+
+						"virtual_name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+
+			"root_block_device": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"delete_on_termination": {
+							Type:     schema.TypeBool,
+							Computed: true,
+						},
+
+						"iops": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+
+						"volume_size": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+
+						"volume_type": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceAwsLaunchConfigurationRead(d *schema.ResourceData, meta interface{}) error {
+	autoscalingconn := meta.(*AWSClient).autoscalingconn
+	ec2conn := meta.(*AWSClient).ec2conn
+
+	if v, ok := d.GetOk("name"); ok {
+		d.SetId(v.(string))
+	}
+
+	describeOpts := autoscaling.DescribeLaunchConfigurationsInput{
+		LaunchConfigurationNames: []*string{aws.String(d.Id())},
+	}
+
+	log.Printf("[DEBUG] launch configuration describe configuration: %s", describeOpts)
+	describConfs, err := autoscalingconn.DescribeLaunchConfigurations(&describeOpts)
+	if err != nil {
+		return fmt.Errorf("Error retrieving launch configuration: %s", err)
+	}
+
+	lc := describConfs.LaunchConfigurations[0]
```

Comment:
To prevent potential panics, we should perform a `nil` check on `describConfs` and `len()` check on `describConfs.LaunchConfigurations`, e.g.

```go
if describConfs == nil || len(describConfs.LaunchConfigurations) == 0 {
  return errors.New("No matching Launch Configuration found")
}

if len(describConfs.LaunchConfigurations) > 1 {
  return errors.New("Multiple matching Launch Configurations found")
}
```